packages/diracx-web-components/tsup.config.ts

```diff
@@ -0,0 +1,61 @@
+import * as fs from "node:fs/promises";
+import * as path from "path";
+import { defineConfig } from "tsup";
+
+export default defineConfig([
+  {
+    entry: [
+      "components/!(index|*.stories).ts?(x)",
+      "components/*/!(index|*.stories).ts?(x)",
+      "hooks/!(index|*.stories).ts?(x)",
+      "contexts/!(index|*.stories).ts?(x)",
+      "types/!(index|*.stories).ts?(x)",
+    ],
+    format: ["esm", "cjs"],
+    experimentalDts: true, // Seems to work fine, lower memory usage and faster than dts
+    target: "es5",
+    bundle: true,
+    sourcemap: true,
```

Comment:
I only minify the code when using `npm build` using --minify because I thought having the compiled code not minified during development  could be useful but it's arguable.

And I didn't look much at the treeshake option because tsup uses eslint's treeshake by default, and this option switches to the rollup one. I don't know how useful it could be. Any opinion?

Reply:
Alright I see, thanks for the details! Let's keep it like it is done now.